vm/vmcore/src/synic.rs

```diff
@@ -101,5 +101,22 @@ pub trait GuestEventPort: Send + Sync {
     fn clear(&mut self);
 
     /// Updates the parameters for the event port.
-    fn set(&mut self, vtl: Vtl, vp: u32, sint: u8, flag: u16);
+    fn set(&mut self, vtl: Vtl, vp: u32, sint: u8, flag: u16) -> anyhow::Result<()>;
+}
+
+/// A guest message port, created by [`SynicPortAccess::new_guest_message_port`].
+pub trait GuestMessagePort: Send + Sync {
+    /// Posts a message to the guest.
+    ///
+    /// It is the caller's responsibility to not queue too many messages. There
+    /// is no backpressure mechanism at the transport layer.
+    ///
+    /// FUTURE: add backpressure.
+    fn post_message(&self, typ: u32, payload: &[u8]);
+
+    /// Changes the virtual processor to which messages are sent.
+    fn set_target_vp(&mut self, vp: u32) -> anyhow::Result<()>;
+
+    /// Gets the current virtual processor to which messages are sent.
+    fn target_vp(&self) -> u32;
```

Comment:
Wondering why we need to be able to query it.

Reply:
Oh, I see, it's for inspect.

Rather than add this, I'd suggest either just remember the target VP you set in vmbus_server and use that for the inspection, or make this trait depend on `Inspect` (or `InspectMut`) and inspect the whole value. I'd probably suggest the latter.